packages/styles/scss/components/buttongroup/_buttongroup.scss

```diff
@@ -57,6 +57,8 @@
     ::slotted([slot='icon']) {
       position: absolute;
       right: 1rem;
+      top: 13px;
+      bottom: 13px;
```

Comment:
We'd want to avoid hard-coding the position, especially considering different size of button. That said, would you be able to try `top: 50%; transform: translateY(-50%)` trick?